### ISSUE TITLE
주제 확정 미팅 내용 (9/3)

### ISSUE THREAD

**Issue by jh0shin** [jh0shin/Graph-Neural-Network#1]
Graph neural network의 성능 향상 목표는 model accuracy 상향 보다는 computation time의 optimization이 목표입니다. 교수님이 말씀하신 이번 연구의 목표는, graph neural network에서는 graph를 matrix로 표현하고 matrix multiplication을 이용하여 연산을 수행하게 되는데, 기존에 존재하는 library source code를 따라가며 low level matrix multiplication code의 최적화입니다.

 Graph neural network 관련 존재하는 library로는 [pytorch geometric(docs)](https://pytorch-geometric.readthedocs.io/en/latest/), [deep graph library](https://www.dgl.ai/)가 있으며, 1주차의 목표는 다음주 수요일(9/8)까지 pytorch geometric 2명, deep graph library 2명으로 나누어져 GCN(Graph Convolution Network)을 아무거나 하나씩 직접 구현하여 돌려보는 것입니다.

 연구의 중심이 될 수 있는 내용은 아래와 같습니다.
- GNN library에서 호출되는 cuda code를 찾고, optimizing sparse matrix multiplicate computation
- Python library에서 cuda code를 호출하는 부분에서의 optimization
- library 없이 cpu에서 graph multiplication의 optimization
